simpegEM1D/Survey.py

```diff
@@ -394,19 +396,19 @@ def lowpass_filter(self):
 
         return self._lowpass_filter
 
-    def set_frequency(self):
+    def set_frequency(self, pts_per_dec=-1):
         """
         Compute Frequency reqired for frequency to time transform
         """
         if self.wave_type == "general":
             _, frequency, ft, ftarg = check_time(
-                self.time_int, 0, 'sin',
-                {'pts_per_dec': 3, 'fftfilt': self.fftfilt}, 0
+                self.time_int, -1, 'dlf',
+                {'pts_per_dec': pts_per_dec, 'dlf': self.fftfilt}, 0
             )
```

Comment:
Great! `pts_per_dec` by default to -1, as we discussed earlier.

Just in case you wonder: You could still define `sin` or `cos` by providing 
`{'kind': 'sin', 'pts_per_dec': pts_per_dec, 'dlf': self.fftfilt}`

However, like this it is perfect, because the internal defaults are:
- step-on: 'sin' (always, enforced)
- step-off: 'cos' (always, enforced)
- impulse: 'sin' by default, but the `kind`-argument overwrites that.

Reply:
Yeah, I found setting pts_per_dec = 3 can make significant errors when we are simulating basement.  Fine to sacrifice speed. 

Ah, great to know. Was following your documentation. For simpegEM1D, I only need step-off, but good to know. Thanks!